src/lib.rs

```diff
@@ -92,11 +92,57 @@ impl<P: Pipe> PythonParserBuilder<P> {
 
     pub fn load_and_init_class<'p>(py: Python<'p>, module_name: &str, class_name: &str, options: &[(String, String)]) -> PyResult<PyObject> {
         let module = try!(Self::load_module(py, module_name));
+        let dict = module.dict(py);
+
+        if dict.contains(py, "error").unwrap() {
+            warn!("Already implemented error function, omitting callback definition.");
+        } else {
+            dict.set_item(py, "error", py_fn!(python_error_callback(error_message: &str))).unwrap();
+        }
+
+        if dict.contains(py, "info").unwrap() {
+            warn!("Already implemented info function, omitting callback definition.");
+        } else {
+            dict.set_item(py, "info", py_fn!(python_info_callback(info_message: &str))).unwrap();
+        }
+
+        if dict.contains(py, "trace").unwrap() {
```

Comment:
The `if dict.contains() else ...` structure is very often used, we can create a function/macro for it.

The documentation [1] says, that `py_fn!` creates a new type (R) which implements `ToPyObject` and returns PyResult<R>.

A function which can deal with this unknown type would look like this:

```
fn inject_function<F: ToPyObject>(py: Python, dict: PyDict, name: &str, function: F) {                                                                             
    if dict.contains(py, name).unwrap() {                                                                                                                          
        warn!("Already implemented {}() function, omitting callback definition.", name);                                                                           
    } else {                                                                                                                                                       
        dict.set_item(py, name, ).unwrap();                                                                                                                        
    }                                                                                                                                                              
}                                                                                                                                                                  
```                                                                                                                                                                   
                                                                                                                                                                   
Call site:

```                                                                                                                                                              
let function = try!(py_fn!(python_warning_callback(warning_message: &str)));                                                                                       
inject_function(py, &mut dict, name, function); 
```

I didn't try it but AFAIK it should work.